Userland/Libraries/LibWebView/URL.cpp

```diff
@@ -16,16 +18,32 @@
 
 namespace WebView {
 
+static Optional<URL> create_url_with_url_or_path(String const& url_or_path)
+{
+    auto url = Unicode::create_unicode_url(url_or_path);
+    if (!url.is_error() && url.value().is_valid())
+        return url.release_value();
+
+    auto path = LexicalPath::canonicalized_path(url_or_path.to_deprecated_string());
+    auto url_from_path = URL::create_with_file_scheme(path);
+    if (url_from_path.is_valid())
+        return url_from_path;
```

Comment:
Why not just continue calling `URL::create_with_url_or_path`?

Reply:
The idea is that `create_url_with_url_or_path` is practically the same as `URL::create_with_url_or_path` just with the URL constructor replaced by `Unicode::create_unicode_url`

Reply:
That looks neat, but the problem I see with it would be that `URL::create_with_url_or_path` still calls the `URL` constructor, so in case there was an issue with the Unicode conversion, it would lead to a "valid" URL containing non-ASCII characters, instead of an empty optional.